src/main/java/com/rabbitmq/client/ChannelContinuationTimeoutException.java

```diff
@@ -0,0 +1,35 @@
+package com.rabbitmq.client;
+
+import java.io.IOException;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Exception thrown when a channel times out on a continuation during a RPC call.
+ * @since 4.1.0
+ */
+public class ChannelContinuationTimeoutException extends IOException {
+
+    /**
+     * The channel that performed the call.
+     */
+    private final Object channel;
```

Comment:
This exception is thrown from `AMQChannel`, which is not a `Channel`. There are several solutions: 
 * having a `Channel` property in the exception and casting the `AMQChannel` to `Channel` when throwing the exception. We risk in theory a `ClassCastException`.
 * having an `AMQChannel` property in the exception, but the implementation leaks up to error handling.
 * having just the channel number and the connection in the exception, which should be enough to diagnose any error.

I think the third solution is a good compromise.